### PR TITLE
Add symlink for current logging directory to actual logging directory

### PR DESCRIPTION
**Pull request by daxy01**
Adding a symlink from "log/<timestamp>" to "log/current" which allows viewing the logfile by accessing "log/current/log.txt".

### CODE REVIEW

**Comment by daxy01** [comment]
To implement https://github.com/Koenkk/zigbee2mqtt/issues/7064

**Comment by daxy01** [comment]
I guess the test fails because of the test-case which expects one directory and the symlink is counted as directory # 2. 
Would it make sense to separate out the symlink creation outside of the log directory creation?

File: lib/util/logger.js [hunk 1]
```diff
@@ -15,10 +15,16 @@ const output = settings.get().advanced.log_output;
...
+const current = settings.get().advanced.log_directory.replace('%TIMESTAMP%', 'current');
+const actual = './' + timestamp;
 
 // Make sure that log directoy exsists when not logging to stdout only
 if (output.includes('file')) {
+    if (fs.existsSync(current)) {
+      fs.unlinkSync(current);
+    }
     fx.mkdirSync(directory);
+    fs.symlinkSync(actual, current);
```
Comment (Koenkk) [inline lib/util/logger.js:27]: Have you tested if this also works on windows?

Comment (daxy01) [reply lib/util/logger.js:27]: I just confirmed this is working on Windows 10 Pro:
```
C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging>npm start

> zigbee2mqtt@1.18.2-dev start C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging
> node index.js

Zigbee2MQTT:info  2021-04-23 14:09:45: Logging to console and directory: 'C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\data\log\2021-04-23.14-09-44' filename: log.txt
Zigbee2MQTT:info  2021-04-23 14:09:45: Starting Zigbee2MQTT version 1.18.2-dev (commit #unknown)
Zigbee2MQTT:info  2021-04-23 14:09:45: Starting zigbee-herdsman (0.13.91)
Zigbee2MQTT:error 2021-04-23 14:09:45: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-04-23 14:09:45: Failed to start zigbee
Zigbee2MQTT:error 2021-04-23 14:09:45: Check https://www.zigbee2mqtt.io/information/FAQ.html#help-zigbee2mqtt-fails-to-start for possible solutions
Zigbee2MQTT:error 2021-04-23 14:09:45: Exiting...
Zigbee2MQTT:error 2021-04-23 14:09:45: Error: Error while opening serialport 'Error: Opening /dev/ttyACM0: Unknown error code 3'
    at Znp.<anonymous> (C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\dist\adapter\z-stack\znp\znp.js:136:32)
    at Generator.next (<anonymous>)
    at C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\dist\adapter\z-stack\znp\znp.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\dist\adapter\z-stack\znp\znp.js:23:12)
    at SerialPort.<anonymous> (C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\dist\adapter\z-stack\znp\znp.js:134:49)
    at SerialPort._error (C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\node_modules\@serialport\stream\lib\index.js:198:14)
    at C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\node_modules\zigbee-herdsman\node_modules\@serialport\stream\lib\index.js:242:12
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.2-dev start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zigbee2mqtt@1.18.2-dev start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Marcel\AppData\Roaming\npm-cache\_logs\2021-04-23T12_09_45_546Z-debug.log

C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging>dir data\log
 Volume in drive C has no label.
 Volume Serial Number is 3845-A2B4

 Directory of C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\data\log

04/23/2021  02:09 PM    <DIR>          .
04/23/2021  02:09 PM    <DIR>          ..
04/23/2021  02:08 PM    <DIR>          2021-04-23.14-08-37
04/23/2021  02:09 PM    <DIR>          2021-04-23.14-09-44
04/23/2021  02:09 PM    <SYMLINKD>     current [.\2021-04-23.14-09-44]
               0 File(s)              0 bytes
               5 Dir(s)  20,619,022,336 bytes free

C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging>dir data\log\current
 Volume in drive C has no label.
 Volume Serial Number is 3845-A2B4

 Directory of C:\Users\Marcel\Documents\zigbee2mqtt-add-current-logging\data\log\current

04/23/2021  02:09 PM    <DIR>          .
04/23/2021  02:09 PM    <DIR>          ..
04/23/2021  02:09 PM               390 log.txt
               1 File(s)            390 bytes
               2 Dir(s)  20,619,022,336 bytes free
```
Note: Only tested the logging part, z2m fails to start as no USB has been inserted.

Comment (Koenkk) [reply lib/util/logger.js]: Thanks, I've made this an option now (defaults to false):

```yaml
advanced:
  log_symlink_current: true
```

Can you check it? Also added a test

Comment (daxy01) [reply lib/util/logger.js]: Excellent, just tested the changes and they do work fine, including the settings.
Out of sheer interest, any specific reason to make this setting false by default?

Comment (Koenkk) [reply lib/util/logger.js]: Because I don't think everyone is interested in this, so better to have it configurable.

**Pull request merged**